api/profilebase/v1alpha1/profilebase.go

```diff
@@ -17,13 +17,68 @@ limitations under the License.
 package v1alpha1
 
 import (
+	"context"
+	"fmt"
+
 	rcommonv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"sigs.k8s.io/security-profiles-operator/api/profilerecording/v1alpha1"
 	secprofnodestatusv1alpha1 "sigs.k8s.io/security-profiles-operator/api/secprofnodestatus/v1alpha1"
 )
 
+const ProfilePartialLabel = "spo.x-k8s.io/partial"
+
+type MergeableProfile interface {
+	client.Object
+
+	Merge(profile MergeableProfile) error
+}
+
+type SecurityProfileBase interface {
+	client.Object
+
+	ListProfilesByRecording(ctx context.Context, cli client.Client, recording string) ([]metav1.Object, error)
+	IsPartial() bool
+}
+
+func IsPartial(obj metav1.Object) bool {
+	_, ok := obj.GetLabels()[ProfilePartialLabel]
+	return ok
+}
+
+func ListProfilesByRecording(
+	ctx context.Context,
+	cli client.Client,
+	recordingName string,
+	recordingNamespace string,
+	list client.ObjectList,
+) ([]metav1.Object, error) {
+	if err := cli.List(
+		ctx,
+		list,
+		client.InNamespace(recordingNamespace),
+		client.MatchingLabels{v1alpha1.ProfileToRecordingLabel: recordingName}); err != nil {
```

Comment:
We might need to pre-parse the `recordingName` to ensure it's usable as a label. Remember labels have a length limitation. We might need to limit recording names or hash in case the length is too long.

Reply:
Good call!

Do you (or @sascha) have a preference? I think I prefer limiting the names, just because the hashes are user-unfriendly and there's really no strict need to keep them unique (unlike e.g. node labels).

An easy way might be to add an event in the webhook recorder that says the recording name is too long and no recording will happen, but I'm not sure if it's super well discoverable. Unfortunately we don't have a profilerecording controller except the merger. Actually, thinking out loud, the merger might be a good place to do that, even if it's not strictly its job. (I'd prefer to keep the webhooks as simple as possible).

This wouldn't prevent the recordings from being created with names that are too long, though we'd need an admission webhook for that which I think is really an overkill..

Reply:
No preferences on my side. I'd love it if we could just leverage the CRD definitions to limit the name size (as we can add validations to fields in the CRD)... but I'm not sure and I don't think that's possible. So... whatever you see fit. All I'd like is that this is taken into account.

Reply:
OK, there's a new comment that emits an event in the webhook as well as error when collecting the recording. Instead of the error we could be more permissive and just create the recording as usual (so not partial) and without the label.

Reply:
> I'm not sure myself what's preferable, but error seems somehow "safer" than guessing.

What would happen if we didn't error out? Would we truncate and continue?